corehq/apps/app_manager/models.py

```diff
@@ -4467,8 +4468,35 @@ def make_build(self, comment=None, user_id=None):
         assert copy._id
         prune_auto_generated_builds.delay(self.domain, self._id)
 
+        self.check_build_dependencies(new_build=copy)
+
         return copy
 
+    def check_build_dependencies(self, new_build):
+        """
+        Reports whether the app dependencies have been removed.
+        """
+
+        def has_dependencies(build):
+            if isinstance(build, Application):
```

Comment:
wont it always be Application object since we wrapped it?
Its unclear why we are checking this here.

Reply:
The `new_build` is wrapped and will be an instance of `Application`, but the `last_build` is not (see [here](https://github.com/dimagi/commcare-hq/blob/1dc56108914ab6fca31be179aa01a931039eddbd/corehq/apps/app_manager/dbaccessors.py#L116)). I suppose I have two options:
1. Also make sure to wrap the `last_build` results and remove the instance checking
2. Leave as is and handle different instance types.

Any preference from your side?

Reply:
Actually, no, I'm going to wrap it as well, because I think the fact that you had to ask this question makes the code less obvious as it is now.

Reply:
https://github.com/dimagi/commcare-hq/pull/35455/commits/1823cbdc9038182a69e0154cad8695db895c0b9d